.changeset/silly-clocks-tan.md

```diff
@@ -0,0 +1,66 @@
+---
+"@fluidframework/tree": major
+---
+
+Implement compatibility-based schema evolution API
+
+This change adjusts some top-level APIs for using SharedTree in a context that may necessitate changing a document's schema.
+First of all, `SharedTree`'s restrictions around view schema and stored schema compatibility have been slightly relaxed:
+A particular view schema can now be used to open documents whose stored schema supports a superset of that view schema.
+Beware that applications which choose to do this may experience runtime incompatibilities in application logic: for instance,
+the stored schema for a document may contain allowed types in fields which the view schema isn't expecting,
+which could result in application errors.
+The motivation for this change is to eventually allow more flexible application policies around what types of schema incompatibilities
+are OK to allow different clients to collaborate using.
+
+Application authors are encouraged to develop a compatibility policy which they are comfortable with using the guidance in the
+"Schema Evolvability" section of `@fluidframework/tree`'s readme.
+
+To make the details of schema compatibilities that SharedTree supports more clear,
+`TreeView.error` has been functionally replaced with the `compatibility` property.
+Users desiring the previous strict behavior can use `view.compatibility.isExactMatch` at appropriate places of application logic.
+
+In addition to this functional change to the types of collaboration that `SharedTree` permits, several APIs have been tweaked.
+
+# `ITree.schematize` deprecation
+
+`ITree.schematize` has been deprecated in favor of `ITree.viewWith`.
+Unlike `schematize`, `viewWith` does not implicitly initialize the document. As such, it doesn't take an `initialTree` property.
+Instead, applications should initialize their trees in document creation codepaths using the added `TreeView.initialize` API.
+
+As an example, something like the following code may have been used before for both the document create and document load codepaths:
+
+```typescript
+const tree = SharedTree.create(runtime, "foo");
+const view = tree.schematize({ schema: Point, initialTree: () => new Point({ x: 0, y: 0 }) });
```

Comment:
This would not work in the before, because it needs to construct a TreeConfiguration (its a class not a interface).

Having this before version be inaccurate in this way makes it hard to evaluate the intended change on applications. Currently apps tend to declare their configuration in the schema file, and the choice of the class based API was made to help them be type safe and improve runtime and compile time errors. I cannot tell if that is regressed by this change as the before and after APis for users like this are not clear.

Reply:
```suggestion
const view = tree.schematize(new TreeConfiguration(Point, () => new Point({ x: 0, y: 0 })));
```